Source/WebCore/loader/ContentFilter.h

```diff
@@ -89,6 +93,24 @@ class ContentFilter : public CanMakeWeakPtr<ContentFilter>, public CanMakeChecke
 private:
     using State = PlatformContentFilter::State;
 
+    class ContentFilterCallbackAggregator : public RefCounted<ContentFilterCallbackAggregator> {
+    public:
+        static auto create(ContentFilter& contentFilter, CompletionHandler<void(ResourceRequest&&)>&& callback) { return adoptRef(*new ContentFilterCallbackAggregator(contentFilter, WTFMove(callback))); }
+
+        virtual ~ContentFilterCallbackAggregator();
+
+        void didReceivePlatformContentFilterDecision(PlatformContentFilter&, ResourceRequest&&, PlatformContentFilter::State);
+
+    private:
+        explicit ContentFilterCallbackAggregator(ContentFilter&, CompletionHandler<void(ResourceRequest&&)>&&);
+
+        CheckedPtr<ContentFilter> m_contentFilter;
+        CompletionHandler<void(ResourceRequest&&)> m_callback;
+        unsigned m_platformFiltersAllowed { 0 };
```

Comment:
I think the name is a little unclear for a `count`. Maybe something like `m_numberOfFiltersAllowed` or `m_filtersAllowedCount`